doc/versioning_and_life_cycle.md

```diff
@@ -0,0 +1,72 @@
+
+The official Azure CLI follows [Semantic Versioning](https://semver.org/) for version numbering. The CLI version consists of three parts `MAJOR.MINOR.PATCH`:
+- MAJOR version is upgraded only when there is a huge revolution. Generally in Azure CLI, this version is not upgraded.
+- MINOR version is upgraded for new features with the normal release process. All feature updates must be forward compatible, and breaking changes can only be published in the breaking change window *(Ignite sprint or Build sprint)*.
+- PATCH version is upgraded for hotfixes through the hotfix release process.
+
+## Life Cycle
+
+### Release States
+
+Azure CLI differs from the [three life cycles](https://docs.microsoft.com/learn/modules/describe-service-life-cycle-microsoft-365/2-private-public-general-availability) of an Azure Service.  CLI only uses two states, **GA** (Generally Available) and **Preview**, making it easier to understand and maintain.
+
+- The **GA** state means that a feature is mature, stable, available to all customers, and has formal support. A GA feature must be forward compatible and not prone to cause breaking changes.
+    - Commands not tagged with the `is_preview` flag are considered as GA commands
+
+- The **Preview** state means a feature is unstable and not finalized. Preview features are released before GA and allow customers to test, experiment, and provide feedback in advance of the GA release. Preview features should not be used in production environments as they aren't stable or finalized and could have breaking changes with each release up to, and including, the feature's GA release.
+    - Preview features and commands need to be tagged with the [is_preview](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#preview-commands-and-arguments) flag to let customers know about instability.
+    - It is generally recommended to put the preview features into CLI extension first and migrate it to official CLI after they are stable and GA. We can also consider moving some preview features from official CLI to CLI extensions if they are proper for moving.
+
+### States Transition
+
+#### Preview to GA
+Generally, if the service is stable and the design of the commands will not change, we are confident that these commands will not cause breaking changes in the future, then we can remove the `is_preview` flag and GA those commands.
+
+#### GA to Preview
+For the official CLI, it is generally not allowed to downgrade from GA to Preview, because the GA state means that the stability of commands has been promised to customers, and new instability cannot be introduced by downgrade to Preview. Moreover, since Azure CLI is an all-in-one package, users usually upgrade directly to the latest version of CLI. If users skip the intermediate version with those feature downgraded to Preview and directly upgrade to the latest version that has breaking changes, it will make a terrible experience for them.
```

Comment:
```suggestion
For the official CLI, it is generally not allowed to downgrade from GA to Preview, because the GA state means that the stability of commands has been promised to customers, and new instability cannot be introduced by downgrading to Preview. Moreover, since Azure CLI is an all-in-one package, users usually upgrade directly to the latest version of CLI. If users skip the intermediate version with features having been downgraded to Preview and directly upgrade to the latest version that has breaking changes, it will make a confusing and frustrating experience for them.
```